3-Authorization-II/protect-web-api/resource-api/src/main/java/com/microsoft/azuresamples/msal4j/msidentityspringbootwebapi/AppSecurityConfig.java

```diff
@@ -0,0 +1,72 @@
+package com.microsoft.azuresamples.msal4j.msidentityspringbootwebapi;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.security.oauth2.resource.OAuth2ResourceServerProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.oauth2.core.DelegatingOAuth2TokenValidator;
+import org.springframework.security.oauth2.core.OAuth2TokenValidator;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtClaimNames;
+import org.springframework.security.oauth2.jwt.JwtClaimValidator;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtIssuerValidator;
+import org.springframework.security.oauth2.jwt.JwtTimestampValidator;
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder;
+import org.springframework.util.StringUtils;
+
+import com.microsoft.azuresamples.msal4j.msidentityspringbootwebapi.experimental.*;
+
+@Configuration
+public class AppSecurityConfig {
+
+    @Value("${azure.activedirectory.client-id}")
+    String clientid;
+    
+    @Value("${azure.activedirectory.tenant-id}")
+    String tenantId;
+    
+    @Value("${azure.activedirectory.instance-uri}")
+    String instanceUri;   
+    
+    /**
+     * JwtDecoder for signature and claim validation 
+     */
+    @Bean
+    JwtDecoder jwtDecoder() {
+    	String JWKSet = instanceUri + tenantId + "/discovery/v2.0/keys";
+        NimbusJwtDecoder nimbusJwtDecoder = NimbusJwtDecoder.withJwkSetUri(JWKSet).build();
+        nimbusJwtDecoder.setJwtValidator(jwtValidator());
+        return nimbusJwtDecoder;
+    }
+    
+    /* 
+     * Adds standard and optional extended validation for AAD token
+     */
+    private OAuth2TokenValidator<Jwt> jwtValidator() {
+        List<OAuth2TokenValidator<Jwt>> validators = new ArrayList<>();
+        
+        // Add standard claim validation for valid issuer, audience, and timestamp claims
+        String issuerUri =  instanceUri + tenantId + "/v2.0";
+        validators = AADHelpers.AADStandardValidators(validators, issuerUri, clientid);
+        
+        /*Extended Validation 
```

Comment:
I would recommend referring to msal-java or microsoft identity platform documentation in these comments, whenever possible. For instance, token validation should refer to token validation documentation for more info. 

You could re-use the comments from .NET 4-1